### PR TITLE
feat(skills): add github-dependabot-maintain skill

### LINKED ISSUE #1050: Dependabot 設定メンテナンス用の Agent Skill を追加する
## Overview / 概要

`.github/dependabot.yml` をメンテナンスするための Agent Skill を `.claude/skills/github-dependabot-maintain/SKILL.md` として新規追加する。

本 Skill は `projects/` 直下のプロジェクトを自動スキャンし、Dependabot エントリの追加・削除を検出してユーザーに diff を提示、承認後に `dependabot.yml` を更新する。

## Objective / 目的

- monorepo 内で新規プロジェクトが増えた際の Dependabot 設定漏れを減らす
- ディレクトリが移動・削除されたりロックファイルがなくなった際の無効エントリ残存を防ぐ
- エントリ順序やグループ名の命名規則を自動で統一し、レビューしやすい diff にする

## Background / 背景

現在の `.github/dependabot.yml` には以下 4 エントリが登録されている。

- `/projects/portfolio`（bun）
- `/projects/file-server`（bun）
- `/projects/edit-vid`（uv）
- `/projects/simple-agent-poc`（uv）

一方、`projects/` 直下には `portal` / `edit-vid2` / `raw-webgl2-shader` など、ロックファイルを持つ未登録の bun プロジェクトも存在する。手動での追記は漏れやすいため、Skill 化して定型的なメンテナンスを自動化する。

## Related Issues/PRs / 関連Issue・PR

None

## Implementation Approach / 実装方針

1. `.claude/skills/github-dependabot-maintain/SKILL.md` を作成する
2. 既存の `dependabot.yml` のパターンをテンプレートとして定義する
3. `projects/` 直下の自動スキャン、ecosystem 判定、エントリ追加・削除、diff 提示・承認、保存後検証のフローを記述する
4. 必要に応じて `.claude/skills/README.md` などの Skills 一覧を更新する

### 判定ロジック

```mermaid
flowchart TD
  Start[ユーザーが dependabot メンテを依頼] --> Scan[projects/ 直下をスキャン]
  Scan --> Lock{ロックファイルあり？}
  Lock -- "bun.lock / bun.lockb" --> Bun[bun エントリ候補]
  Lock -- "uv.lock" --> Uv[uv エントリ候補]
  Lock -- "なし" --> Skip[対象外]
  Bun --> Compare[既存 dependabot.yml と比較]
  Uv --> Compare
  Compare --> Add[不足エントリ追加]
  Compare --> Remove["directory / ロックファイル不在の無効エントリ削除"]
  Compare --> Align[グループ名を命名規則に統一]
  Add --> Sort[directory アルファベット順にソート]
  Remove --> Sort
  Align --> Sort
  Sort --> Diff[diff を提示]
  Diff --> Approve{ユーザー承認？}
  Approve -- "Yes" --> Save[dependabot.yml 保存]
  Approve -- "No" --> Abort[中断]
  Save --> Validate[PyYAML 構文検証 + カバレッジ再確認]
```

### 新規エントリのテンプレート

```yaml
- package-ecosystem: "bun"  # または "uv"
  directory: "/projects/{project-name}"
  schedule:
    interval: "weekly"
  open-pull-requests-limit: 1
  rebase-strategy: "disabled"
  groups:
    {project-name}-minor-and-patch:
      applies-to: version-updates
      patterns:
        - "*"
      update-types:
        - "minor"
        - "patch"
```

### 重要な制約

- スキャン対象は `projects/` の直下のみ。`_labs/` / `_samples/` / ネストしたサブディレクトリは対象外
- ecosystem はロックファイル優先で判定する
  - `bun.lock` または `bun.lockb` があれば `bun`
  - `uv.lock` があれば `uv`
- 既存エントリの `schedule` / `open-pull-requests-limit` / `rebase-strategy` は変更しない
- 書き換え前に `.github/dependabot.yml` の未コミット変更を確認し、あれば警告して停止する
- 保存後は PyYAML で構文検証し、スキャン対象プロジェクトとエントリが 1:1 か再確認する

## Tasks / タスク

- [ ] `.claude/skills/github-dependabot-maintain/SKILL.md` を作成する
- [ ] Skill の動作確認（dry-run 的な手動検証）を行う
- [ ] `.claude/skills/README.md` 等の Skills 一覧を更新する

## Out of Scope / スコープ外

- Dependabot のスケジュールや `rebase-strategy` などの戦略変更
- `github-actions` や `docker` など他の package-ecosystem 対応
- security updates 設定の変更
- 自動での PR 作成やマージ

### Validation Policy / 検証方針

Skill 作成後、実際の `dependabot.yml` を使って手動検証を行う。具体的には、Skill で定義したフローに従って既存エントリを維持しつつ未登録プロジェクトを追加するシミュレーションを行い、YAML 構文とカバレッジを確認する。

### Documentation / ドキュメント

- `.claude/skills/README.md` 等の Skills 一覧があれば更新する
- 本 Issue の対象外だが、必要に応じて `.github/dependabot.yml` 自体のコメント追記も検討する

## Acceptance Criteria / 完了条件

- [ ] `.claude/skills/github-dependabot-maintain/SKILL.md` が追加されている
- [ ] Skill 内のフローに従って `dependabot.yml` をメンテできる
- [ ] YAML 構文検証とカバレッジ再確認が定義されている
- [ ] Skills 一覧が更新されている

## Test Perspectives / テスト観点

| Area / 観点 | What to Verify / 確認内容 | Method / 確認方法 |
| --- | --- | --- |
| Normal path / 正常系 | 既存 4 エントリが維持され、未登録プロジェクトが追加される | `dependabot.yml` diff 確認 |
| Edge cases / 境界条件 | ロックファイルのないディレクトリ、`_labs` / `_samples` が対象外となる | スキャン結果の確認 |
| Regression / 回帰 | 既存エントリの `schedule` / `open-pull-requests-limit` / `rebase-strategy` が変更されない | diff 確認 |
| Operations / 運用 | YAML 構文が有効であり、dependabot 設定として解釈可能である | `python3 -c "import yaml; yaml.safe_load(...)"` |

## Notes / 補足

- グループ名は `{project-name}-minor-and-patch` に統一する
- diff 提示後、ユーザーが承認した場合のみファイルを保存する
- 保存後の検証で 1:1 でなくなった場合はエラーを報告し、手動修正を促す

### PR DESCRIPTION
**Pull request by u7chan**
## Issues

- Close #1050

## Why

Dependabot 設定を `projects/` の実際の構成と同期させる運用を Skill 化し、新規プロジェクト追加や無効エントリの削除を自動検出する。

## Summary

monorepo 専用の `.claude/skills/github-dependabot-maintain/SKILL.md` を追加する。
ロックファイルから ecosystem を判定し、不足・削除対象のエントリを diff 提示して承認後に更新するフローを定義する。
合わせて monorepo 内の既存 Skill 一覧を `.claude/skills/README.md` にまとめる。

## Changes

- Add `.claude/skills/github-dependabot-maintain/SKILL.md`
- Add `.claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md`
- Add `.claude/skills/README.md` listing monorepo skills

## Verification

- `bash /home/u7dev/.claude/skills/scripts/validate-skills.sh` — passed (new SKILL.md is within the 180-line limit)
- Dry-run of the skill workflow against current `projects/` and `.github/dependabot.yml` — produced the expected diff (adds `edit-vid2` and `portal`, keeps existing entries, sorts alphabetically)
- `python3 -c "import yaml; yaml.safe_load(open('desired.yml'))"` on the generated desired config — passed

### CODE REVIEW

File: .claude/skills/github-dependabot-maintain/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+| `bun.lock` or `bun.lockb` | `bun` |
+| `uv.lock` | `uv` |
+
+## Step Details
+
+### Step 1: Check for Uncommitted Changes
+
+Before editing, run:
+
+    git diff -- .github/dependabot.yml
```
Comment (u7chan) [inline .claude/skills/github-dependabot-maintain/SKILL.md]: [must] ⚠️ この確認だと unstaged diff しか見ないため、`.github/dependabot.yml` に staged 済みの未コミット変更がある場合でも処理が進み、後続の書き込みでその変更を上書きできます。「uncommitted changes がないこと」を保証するなら、`git diff --cached -- .github/dependabot.yml` も確認するか、`git status --porcelain -- .github/dependabot.yml` で staged/unstaged の両方を止める必要があります。

AIレビュー補助（Codex / GPT-5）によるコメントです 🤖

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  # Sort by project name for deterministic output
+  first=true
+  printf '%s\n' "${pairs[@]}" | sort | while read -r name eco; do
+    if [ "$first" = true ]; then
+      first=false
+    else
+      echo ""  # blank line between entries, matching current format
+    fi
+    cat << ENTRY
```
Comment (u7chan) [inline .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: [must] ⚠️ ここで全エントリをデフォルト値から再生成しているため、`SKILL.md` の Step 4 で求めている既存エントリの `schedule`、`open-pull-requests-limit`、`rebase-strategy`、その他 custom fields の保持が実現できません。このスクリプトをそのまま使うと既存のDependabot設定を黙って削除する可能性があるので、既存YAMLを読み込んで該当entryの設定をmergeする手順にする必要があります。

AIレビュー補助（Codex / GPT-5）によるコメントです 🤖

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+#   cp "$TMP_DIR/desired.yml" "$CONFIG"
+#   python3 -c "import yaml; yaml.safe_load(open('$CONFIG'))"
+```
+
+## After approval
+
+Write the file and validate:
+
+```bash
+cp "$TMP_DIR/desired.yml" "$CONFIG"
```
Comment (u7chan) [inline .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: [must] ⚠️ preview script を通常のスクリプトとして実行すると、終了時に `trap 'rm -rf "$TMP_DIR"' EXIT` で `desired.yml` が削除され、さらに `TMP_DIR` 変数も呼び出し元には残らないため、この after-approval 手順はそのままでは実行できません。承認後に書き込ませるなら、スクリプト内で承認待ちしてから cleanup するか、`desired.yml` を明示的な永続パスへ出力する必要があります。

AIレビュー補助（Codex / GPT-5）によるコメントです 🤖

**Review by u7chan — COMMENTED** [review]
AIレビュー補助（Codex / GPT-5）によるレビューです。

File: .claude/skills/github-dependabot-maintain/SKILL.md (hunk 1 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/SKILL.md]: 対応しました。

SKILL.md の Step 1 を `git status --porcelain -- .github/dependabot.yml` に変更し、staged / unstaged の両方を検出するようにしました。

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md (hunk 2 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: 対応しました。

リファレンススクリプトをテキストベースのブロックパースに書き換え、既存エントリの `schedule` / `open-pull-requests-limit` / `rebase-strategy` などの設定を保持し、`package-ecosystem` / `directory` / `groups` だけを正規化するようにしました。dry-run の diff でも既存エントリの内容が変わらないことを確認しています。

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md (hunk 3 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: 対応しました。

preview を一時ディレクトリではなく `.github/dependabot.yml.preview` という明示的なパスに書き出し、承認後に実ファイルへコピーしてから preview を削除するようにしました。承認前は preview が残るため、after-approval 手順がそのまま実行できます。

File: .claude/skills/github-dependabot-maintain/SKILL.md (hunk 1 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/SKILL.md]: 再チェックしました。`git status --porcelain -- .github/dependabot.yml` に変更されており、staged / unstaged の両方を検知できるため、この指摘は解消されています。

AIレビュー補助（Codex / GPT-5）による再チェックコメントです 🤖

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md (hunk 2 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: 再チェックしました。新規追加された `maintain-dependabot.py` が既存entryを読み込み、既存ディレクトリでは `normalize_block` で `package-ecosystem` / `directory` / `groups` 以外の設定を保持する形になっているため、この指摘は解消されています。

AIレビュー補助（Codex / GPT-5）による再チェックコメントです 🤖

File: .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md (hunk 3 above)
Comment (u7chan) [reply .claude/skills/github-dependabot-maintain/references/github-dependabot-maintain-detailed.md]: 再チェックしました。preview が `.github/dependabot.yml.preview` に出力され、承認後に同じスクリプト内で適用・削除される流れになっているため、この指摘は解消されています。

AIレビュー補助（Codex / GPT-5）による再チェックコメントです 🤖
